### ISSUE TITLE
Doesn't re-render if props aren't updated

### ISSUE THREAD

**Issue by murphyrandle** [tajo/react-portal#10]
We were having a problem today where we needed the component inside of the portal to receive props and re-render.

It looks like this is where portal triggers a re-render:
https://github.com/tajo/react-portal/blob/master/lib/portal.js#L41

And it appears only to happen if props have changed.

**Comment by tajo** [comment]
> Doesn't re-render if props aren't updated

That's actually what you should aim to. There is `shouldComponentUpdate` implemented as a PureRenderMixin (it's a must have optimization assuming that your props and states are immutable), so maybe the problem is there. 

Or can you please be more specific and send some demo (or the exact way how to reproduce it).
Thanks.

**Comment by javiercr** [comment]
I am having a similar problem. Correct me if I am wrong but the current implementation only re-renders if props were changed AND the portal was not open already:

    componentWillReceiveProps(newProps) {
      if (newProps.isOpened !== 'undefined' && this.newProps !== this.props) {
        if (newProps.isOpened && !this.state.active) {
          this.openPortal();
        }
        if (!newProps.isOpened && this.state.active) {
          this.closePortal();
        }
      }
    }


Note that `openPortal()` is the only method that is actually calling `renderPortal()`.

This doesn't reflect the [original implementation](https://gist.github.com/ryanflorence/ac5174cf97dd596e949c) of portals suggested by Ryan Florence, where he always re-renders when props changes:

    componentWillReceiveProps: function(newProps) {
      this._renderPortal(newProps);
    },

Is there any reason to avoid re-rendering if the portal is already open? Thanks!

**Comment by Lendar** [comment]
I agree with @javiercr approach.

`componentWillReceiveProps` should always call either of:

- `renderPortal`
- `openPortal`
- `closePortal`.

---

By the way, is this a typo?

    this.newProps

There are no `this.newProps` elsewhere.

**Comment by javiercr** [comment]
@Lendar good catch, I'd bet that's a typo. @tajo can you confirm it?

**Comment by tajo** [comment]
@javiercr @Lendar Thanks!
I just released `0.6.0` which fixes `componentWillReceiveProps`. It was quite broken. The new implementation: 

```js
componentWillReceiveProps(newProps) {
  // portal's 'is open' state is handled through the prop isOpened
  if (typeof newProps.isOpened !== 'undefined') {
    if (newProps.isOpened) {
      if (this.state.active) {
        this.renderPortal(newProps);
      } else {
        this.openPortal();
      }
    }
    if (!newProps.isOpened && this.state.active) {
      this.closePortal();
    }
  }

  // portal handles its own 'is open' state
  if (typeof newProps.isOpened === 'undefined' && this.state.active) {
    this.renderPortal(newProps);
  }
}
``` 
I know it is a little cryptic. The first part is for the situation when somebody uses `isOpened` portal prop to open/close portal - that's why there is also `closePortal()` and `openPortal()`. The second part is for more common usage when portal takes care about the open/close state by itself. Notice that only `renderPortal()` is needed.

Makes sense? (If so, I'll close this issue soon.)

`this.newProps` was really non-sense typo.

@murphyrandle Does this solve your problem?
